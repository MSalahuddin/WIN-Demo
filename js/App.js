/* eslint-disable no-nested-ternary */
import React, { useEffect} from 'react';
import { StatusBar, LogBox } from 'react-native'
import AsyncStorage from '@react-native-community/async-storage';
import { GetSocial } from 'getsocial-react-native-sdk';
import UserProvider, { UserContext } from './context/User.context';


import PopupProvider from './context/Popup.context';
import BackgroundMusicProvider from './context/BackgroundMusic.context';
import Navigation from './components/navigation';
import {
  LOCAL_STORAGE_NAME,
} from './constants';

if (__DEV__) {
  import('./ReactotronConfig');
}

LogBox.ignoreLogs(['Warning: ...']); // Ignore log notification by message
LogBox.ignoreAllLogs(); //Ignore all log notifications


function App() {
  // We don't use Async Storage directly. This message came from Reactotron which would
  // only appear in dev


  const saveReferrerPayLoad = async (payload) => {
    await AsyncStorage.setItem(LOCAL_STORAGE_NAME.REFERRER_DATA, JSON.stringify(payload));
  }

  const getSocialHandler = () => {
    GetSocial.registerForPushNotifications()
    GetSocial.onNotificationReceived((notification, wasClicked) => {
      try {
        const data = JSON.parse(JSON.stringify(notification.action))
        const DATA = JSON.parse(data)
        saveReferrerPayLoad(DATA)
        if (wasClicked) {
          // click silently
        }
      }
      catch (e) {
        // catch silently
      }
    });
  }


  useEffect(() => {
    GetSocial.init()
    getSocialHandler()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <BackgroundMusicProvider>

      <PopupProvider>
        <UserProvider>

          <StatusBar hidden />
          <Navigation />
        </UserProvider>
      </PopupProvider>

    </BackgroundMusicProvider>
  );
}

export default App;
