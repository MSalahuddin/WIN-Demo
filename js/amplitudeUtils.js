import RNAmplitude from 'react-native-amplitude-analytics';
import Config from 'react-native-config';

export const amplitude = new RNAmplitude(Config.AMPLITUDE_API_KEY);
export const logEvent = (eventName, eventProperties = null) => amplitude.logEvent(eventName, eventProperties);
export const setUserId = userId => amplitude.setUserId(userId);
export const setUserProperties = userProperties => amplitude.setUserProperties(userProperties);
export const increaseUserProperty = (property, amount = 1) => amplitude.addToUserProperty(property, amount);
export const logRevenueV2 = (revenueProperties) => amplitude.logRevenueV2(revenueProperties);