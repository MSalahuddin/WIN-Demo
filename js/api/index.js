/* eslint-disable max-len */
/* istanbul ignore file */
// import { Platform } from 'react-native';
import client from './client';
import { WW_API_ENDPOINTS } from '../constants';


  const playerChallengeRank = (challengeId, challengeTypeEnum) => client.get(WW_API_ENDPOINTS.PLAYER_CHALLENGES_RANK(challengeId, challengeTypeEnum));
const getMachines = (params) => client.get(WW_API_ENDPOINTS.GET_GAMEROOM_MACHINES,{ params: { ...params, includeInactive : false } });
const getMachinesPrizes = (machineId, countryId) => 
  client.get(WW_API_ENDPOINTS.GET_GAMEROOM_MACHINES_PRIZES(machineId, countryId));
const getMachinesSocketBaseUrl = () => client.get(WW_API_ENDPOINTS.GET_MACHINES_SOCKET_BASE_URL);
const getUpcomingTrivia = () => client.get(WW_API_ENDPOINTS.GET_UPCOMING_TRIVIA);
const getDigitalGame = (liveArcade = false,isWinnerCircle = false) => client.get(WW_API_ENDPOINTS.GET_DIGITAL_GAME(liveArcade,isWinnerCircle));
const getQRCode = () => client.get(WW_API_ENDPOINTS.GET_QR_CODE);
const getQuickAddTokens = () => client.get(WW_API_ENDPOINTS.GET_QUICK_ADD_TOKENS);
const postTransferNFT = (data) => client.post(WW_API_ENDPOINTS.POST_TRANSFER_NFT, data);
const getDigitalGameGateway = () => client.get(WW_API_ENDPOINTS.GET_DIGITAL_GAME_GATEWAY);
const putRemovePlayer = () => client.put(WW_API_ENDPOINTS.REMOVE_PLAYER);
const getTriviaWinners = (triviaId) => client.get(WW_API_ENDPOINTS.GET_TRIVIA_WINNERS(triviaId));
const getTriviaById = (triviaId) => client.get(WW_API_ENDPOINTS.GET_TRIVIA_BY_ID(triviaId));







const api = {
  playerChallengeRank,
  getMachines,
  getMachinesPrizes,
  getMachinesSocketBaseUrl,
  getDigitalGame,
  getQRCode,
  getQuickAddTokens,
  postTransferNFT,
  getDigitalGameGateway,
  putRemovePlayer,
  getUpcomingTrivia,
  getTriviaWinners,
  getTriviaById
};

export default api;
