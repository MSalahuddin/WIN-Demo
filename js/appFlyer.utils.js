import { Platform } from 'react-native';
import appsFlyer from 'react-native-appsflyer';
import Config from 'react-native-config';
import { requestTrackingPermission } from 'react-native-tracking-transparency';

const config = {
    devKey: Config.APPSFLYER_DEV_KEY,
    isDebug: false,
    appId: Config.APP_STORE_ID,
    onInstallConversionDataListener: true,
    onDeepLinkListener: true,
    timeToWaitForATTUserAuthorization: 60
};

export const AFInit = async () => {
    try {
        await appsFlyer.initSdk(config);
        await requestTrackingPermission()
        return Promise.resolve()
    } catch (error) {
        return Promise.resolve()
    }
}

export function AFLogCustomEvent(name, values) {
    appsFlyer.logEvent(name, values, (res) => {
    }, (error) => {
    });
}

export function AFsetUserIdentityEvent(userId) {
    appsFlyer.setCustomerUserId(userId, (res) => {
    },(error) => {
    });
}

export const AFLogStandardEvent = async (eventName, eventProperties = null) => {
    appsFlyer.logEvent(eventName, eventProperties, (res) => {
    }, (error) => {
    });
}
