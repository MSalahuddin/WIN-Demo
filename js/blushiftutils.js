import { NativeModules } from 'react-native'

const { BlueshiftBridge } = NativeModules;


export const logBlushiftEvent = (eventName, eventProperties = null) => BlueshiftBridge.trackCustomEvent(eventName, eventProperties, false)
export const setBlueShiftCustomerInfo = (playerId) => BlueshiftBridge.setUserInfoCustomerId(playerId)
export const setBlueShiftCustomerEmailId = (email) => BlueshiftBridge.setUserInfoEmailId(email)
