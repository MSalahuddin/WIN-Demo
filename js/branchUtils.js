import branch, { BranchEvent } from 'react-native-branch';

export const setUserIdentity = async (userId) => {
    branch.setIdentity(userId);
}
export const logBranchCustomEvent = async (eventName, eventProperties = null) => {
    new BranchEvent(eventName, null, eventProperties).logEvent();
}   
export const logBranchStandardEvent = async (eventName, eventProperties = null) => {
  const event =  new BranchEvent(eventName, null, eventProperties);
  event.logEvent();
 }