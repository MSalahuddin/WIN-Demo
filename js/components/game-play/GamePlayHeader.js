import React, { useContext, useState, useEffect, useRef, useImperativeHandle, forwardRef } from 'react';
import { Animated, BackHandler } from "react-native";
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { hasNotch } from 'react-native-device-info';

import IconButton from '../common/IconButton';
import Ellipse from '../common/Ellipse';
import QueueBar from './QueueBar';
import QueueItem from './QueueItem';
import { SOUNDS } from '../../soundUtils';
import { logEvent } from '../../amplitudeUtils';
import { scale, scaleHeight } from '../../platformUtils';
import { gamePlayStrings } from '../../stringConstants';
import { color } from '../../styles';
import { BackgroundMusicContext } from '../../context/BackgroundMusic.context';
import { UserContext } from '../../context/User.context';
import { coin, redLight, redBack, backInactive } from '../../../assets/images';
import { SCREENS, PLAYER_STATUS, ANALYTICS_EVENTS, ANALYTICS_PROPERTIES } from '../../constants';

const Header = styled.View`
  flex-direction: column;
  margin-top: ${hasNotch() ? scaleHeight(24) : 0};
  background-color: transparent;
`;

const MainRow = styled.View`
  flex-direction: row;
  justify-content: space-between;
  padding-right: ${scale(16)};
`;

const SecondaryRow = styled.View`
  flex-direction: row;
  justify-content: flex-end;
  padding-horizontal: ${scale(20)};
  margin-top: ${scale(16)};
`;

const LiveContainer = styled.View`
  border-radius: ${scaleHeight(20)};
  height: ${scaleHeight(40)};
  padding-horizontal: ${scale(5)};
  background-color: ${color.whiteTransparent};
`;

const EllipseWrapper = styled.View`
  margin-top: ${scale(24)};
`;

const GamePlayHeader = ({ navigation, playerStatus, watchers, queueLength, machineData, setShowQuitAlert, challengeIndex, selectedChallenge  }, ref) => {
  const { tokens } = useContext(UserContext);
  const [isBackButtonDisabled, setIsBackButtonDisabled] = useState(false);
  const { playSoundEffect } = useContext(BackgroundMusicContext);
  const isStatusWatchingOrQueue = playerStatus === PLAYER_STATUS.WATCHING || playerStatus === PLAYER_STATUS.IN_QUEUE;
  const fadeAnim = useRef(new Animated.Value(0)).current;
  useEffect(() => {
    // if status is playing and go back button is disabled,
    // this means the back button was pressed during the countdown
    const backButtonPressedInCountdown = playerStatus === PLAYER_STATUS.PLAYING && isBackButtonDisabled;

    if (backButtonPressedInCountdown) {
      setShowQuitAlert(true);
      setIsBackButtonDisabled(false);
    } else if (playerStatus === PLAYER_STATUS.NEXT_PLAYER || playerStatus === PLAYER_STATUS.IN_QUEUE) {
      setIsBackButtonDisabled(false);
    } else if (playerStatus === PLAYER_STATUS.GAME_ENDED) {
      setIsBackButtonDisabled(true);
    }

    const backAction = () => { 
      // eslint-disable-next-line no-use-before-define
      backOnPress(); 
      return true 
    };

    const backHandler = BackHandler.addEventListener("hardwareBackPress", backAction);

    return () => backHandler.remove();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [playerStatus]);

  useEffect(() => {
    // Animated.timing(fadeAnim, {
    //   toValue: 1,
    //   duration: 250,
    // }).start();
  }, []);

  const fadeOut = () => {
    setIsBackButtonDisabled(true);
    // Animated.timing(
    //   fadeAnim,
    //   {
    //     toValue: 0, 
    //     duration: 250,
    //   }
    // ).start();
  }


  const backOnPress = () => {
    const { navigate } = navigation;
    playSoundEffect(SOUNDS.NEGATIVE_POPUP);
    switch (playerStatus) {
      case PLAYER_STATUS.IN_QUEUE:
        logEvent(ANALYTICS_EVENTS.LEFT_MACHINE_QUEUE, {
          [ANALYTICS_PROPERTIES.MACHINE_ID]: machineData.machineId
        });
        setShowQuitAlert(true);
        break;
      case PLAYER_STATUS.WATCHING:
        if(!!selectedChallenge && challengeIndex != 'undefine'){
          navigate(SCREENS.CHALLENGE_GAME, {challengeIndex, selectedChallenge })
        } else {
          navigate(SCREENS.GAME_ROOM);
        }
        break;
      case PLAYER_STATUS.NEXT_PLAYER:
        setIsBackButtonDisabled(true);
        break;
      default:
        setShowQuitAlert(true);
    }
  };

  useImperativeHandle(ref, () => ({
    hideComponents: () => {
      fadeOut();
    }
  }));

  return (
    <Header>
      <MainRow>
        {/* <Animated.View
          style={
            {
              opacity: fadeAnim
            }
          }> */}
          <IconButton
            onPress={backOnPress}
            icon={isBackButtonDisabled ? backInactive : redBack}
            disabled={isBackButtonDisabled}
          />
        {/* </Animated.View> */}
        {/* <Animated.View
          style={
            {
              opacity: fadeAnim
            }
          }> */}
          <EllipseWrapper>
            <Ellipse
              testID="coin-button"
              width={95}
              borderRadius={44}
              icon={coin}
              amount={tokens}
              isGamePlay
              backgroundColor={color.whiteTransparent}
              disabled={!isStatusWatchingOrQueue}
              onPress={() => {
                const { navigate } = navigation;
                navigate(SCREENS.GAME_CARD_RELOAD);
              }}
            />
          </EllipseWrapper>
        {/* </Animated.View> */}
        {/* <Animated.View
          style={
            {
              opacity: fadeAnim
            }
          }> */}
          <QueueBar watchingQueue={watchers} gameQueue={queueLength} />
        {/* </Animated.View> */}
      </MainRow>
      {/* <Animated.View
        style={
          {
            opacity: fadeAnim
          }
        }> */}
        <SecondaryRow>
          <LiveContainer>
            <QueueItem icon={redLight} title={gamePlayStrings.live} />
          </LiveContainer>
        </SecondaryRow>
      {/* </Animated.View> */}
    </Header>
  );
};

GamePlayHeader.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  playerStatus: PropTypes.string.isRequired,
  watchers: PropTypes.number.isRequired,
  queueLength: PropTypes.number.isRequired,
  machineData: PropTypes.shape({
    machineId: PropTypes.number.isRequired
  }).isRequired,
  setShowQuitAlert: PropTypes.func.isRequired
};

// export default GamePlayHeader;

const ControlPanelHeaderForwardRef = forwardRef(GamePlayHeader);


export default ControlPanelHeaderForwardRef
