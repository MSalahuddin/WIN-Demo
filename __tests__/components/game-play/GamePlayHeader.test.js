import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import MockProvider from '../../MockProvider';
import GamePlayHeader from '../../../js/components/game-play/GamePlayHeader';
import { PLAYER_STATUS } from '../../../js/constants';

describe('Component GamePlayHeader', () => {
  const navigation = { pop: jest.fn(), navigate: jest.fn() };
  const playerStatus = PLAYER_STATUS.WATCHING;
  const machineData = { machineId: 0 };
  const setShowQuitAlert = jest.fn();

  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <GamePlayHeader
          navigation={navigation}
          isPlaying
          isInQueue
          watchers={0}
          queueLength={0}
          playerStatus={playerStatus}
          machineData={machineData}
          setShowQuitAlert={setShowQuitAlert}
        />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('navigation.navigate is called when token button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <GamePlayHeader
          navigation={navigation}
          isPlaying={false}
          isInQueue={false}
          watchers={0}
          queueLength={0}
          playerStatus={playerStatus}
          machineData={machineData}
          setShowQuitAlert={setShowQuitAlert}
        />
      </MockProvider>
    );
    fireEvent(getByTestId('coin-button'), 'press');
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
