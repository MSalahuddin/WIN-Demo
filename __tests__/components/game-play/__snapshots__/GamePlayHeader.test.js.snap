// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GamePlayHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "column",
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 24,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 12,
          "left": 12,
          "right": 12,
          "top": 12,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 24,
          "marginTop": 48,
          "opacity": 1,
        }
      }
      testID="icon-button"
    >
      <Image
        opacity={1}
        resizeMode="contain"
        size={40}
        source={1}
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 36,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 44,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "opacity": 0.7,
            "width": 142.5,
          }
        }
        testID="coin-button"
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 36,
                "marginRight": 7.5,
                "width": 36,
              },
            ]
          }
        />
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#000000"
          fontFamily="MEDIUM"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XXSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Calibre-Medium",
                "fontSize": 21,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "marginTop": 10.5,
              },
            ]
          }
        >
          0
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#72D59E",
              "borderRadius": 13.5,
              "height": 27,
              "opacity": 1,
              "paddingLeft": 6,
              "paddingTop": 3,
              "position": "absolute",
              "right": -6.75,
              "width": 27,
            }
          }
          testID="coin-button"
        >
          <Text
            alignCenter={false}
            allowFontScaling={false}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="NORMAL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 30,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 40,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "space-around",
            "marginTop": 36,
            "minWidth": 150,
            "opacity": 0.7,
            "paddingHorizontal": 7.5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "justifyContent": "center",
              "marginHorizontal": 7.5,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 36,
                  "width": 36,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 14,
                },
              ]
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "justifyContent": "center",
              "marginHorizontal": 7.5,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 36,
                  "width": 36,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 14,
                },
              ]
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginTop": 24,
          "paddingHorizontal": 30,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 40,
            "height": 80,
            "opacity": 0.7,
            "paddingHorizontal": 7.5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "justifyContent": "center",
              "marginHorizontal": 7.5,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 36,
                  "width": 36,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 4.5,
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 14,
                },
              ]
            }
          >
            Live
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
